x/auction/keeper/abci.go

```diff
@@ -50,5 +64,34 @@ func (k Keeper) EndBlocker(ctx sdk.Context) {
 			}
 		}
 	}
-	// TODO post MVP (pbal): prune bids and auctions
+
+	//  Prune bids and auctions -- keep last inactive auction per denom (+ bids) at minimum -- PLUS auctions still in the param freshness window
+	denomsFound := make(map[string]bool)
+	blocksToNotPrune := k.GetParamSet(ctx).BlocksToNotPrune
+	auctionsToDelete := make(map[uint32]bool)
```

Comment:
For a data structure that is going to be iterated through and doesn't require random access, this can just be a slice rather than a map.